cpp/source/post_processor.cc

```diff
@@ -25,35 +26,107 @@ template class New_SuperCapacitorPostprocessor<2>;
 template class New_SuperCapacitorPostprocessorParameters<3>;
 template class New_SuperCapacitorPostprocessor<3>;
 
-void compute_energy(std::vector<std::string> const &capacitor_state,
-                    std::vector<double> const &time,
+namespace internal
+{
+class energy_odeint
+{
+public:
+  energy_odeint(std::vector<double> const &time,
+                std::vector<double> const &power);
+
+  void operator()(std::vector<double> const &, std::vector<double> &dxdt,
+                  double const current_time);
+
+private:
+  std::vector<double> const time;
+  std::vector<double> const power;
+};
+
+energy_odeint::energy_odeint(std::vector<double> const &time,
+                             std::vector<double> const &power)
+    : time(time), power(power)
+{
+}
+
+void energy_odeint::operator()(std::vector<double> const &,
+                               std::vector<double> &dxdt,
+                               double const current_time)
+{
+  unsigned int pos = 0;
+  for (unsigned int i = 0; i < time.size(); ++i, ++pos)
+  {
+    if (time[i] >= current_time)
+      break;
+  }
+  if (pos == 0)
+    dxdt[0] = power[0];
+  else
```

Comment:
Yes, but because we use dopri the time step is much larger that the time step that we have between our data point. So, we can still get a good order of convergence.